localstack-core/localstack/services/lambda_/event_source_mapping/pollers/dynamodb_poller.py

```diff
@@ -74,6 +74,18 @@ def extra_metadata(self) -> dict:
             "eventVersion": "1.1",
         }
 
+    def get_records(self, shard_iterator: str) -> dict:
+        try:
+            return super().get_records(shard_iterator)
+        except self.source_client.exceptions.TrimmedDataAccessException as e:
```

Comment:
I think this is not going to be caught because this is a subclass from `ClientError`, and the parent class is already catching that, and if not a "recognized" exception, raised `PipeInternalError from e`. Not sure if we can catch that somehow? Ugh, sorry for this one... 😓

Reply:
I guess we could try to catch `PipeInternalError`, verify if it is raised from the exception we want (not even sure how to do that 👀 ) and if not, re-raise?  Not great... thanks, DynamoDB-local 😅